### PR TITLE
`mask` and `where` should accept a `callable`

### PR DESCRIPTION
**Pull request by j-bennet**
Fixes a bug introduced in https://github.com/dask/dask/pull/10163.

- [x] Closes [#xxxx](https://github.com/dask/dask/issues/10282)
- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

File: dask/dataframe/core.py [hunk 1]
```diff
@@ -3321,10 +3321,11 @@ def _validate_condition(self, cond):
             or is_dataframe_like(cond)
             or is_series_like(cond)
             or is_index_like(cond)
+            or callable(cond)
```
Comment (phofl) [inline dask/dataframe/core.py:3324]: We could check whether the result of the callable can be aligned, but no strong opinion
Otherwise lgtm

Comment (j-bennet) [reply dask/dataframe/core.py]: I added the check, but if it breaks other tests, I'd rather remove it and not complicate things.

File: dask/dataframe/core.py [hunk 2]
```diff
@@ -3316,16 +3316,26 @@ def cummin(self, axis=None, skipna=True, out=None):
...
-                f" which includes Dask or pandas collections, DataFrames or Series."
+                f" which includes Dask or pandas collections, DataFrames or Series, or a Callable."
             )
+        if callable(cond):
+            result = cond(self._meta)
+            if not is_alignable_collection(result):
+                raise ValueError(
+                    f"Condition should be an object that can be aligned with {self.__class__}, "
+                    f" which includes Dask or pandas collections, DataFrames or Series, or a Callable."
+                )
```
Comment (Oreilles) [inline dask/dataframe/core.py:3338]: You could save a few loc and avoid repeating the error with something like:
```py
cond_result = cond(self._meta) if callable(cond) else cond
if not is_alignable_collection(cond_result):
    ...
```

Comment (j-bennet) [reply dask/dataframe/core.py:3338]: Good point, will do.

File: dask/dataframe/core.py [hunk 3]
```diff
@@ -3316,15 +3316,17 @@ def cummin(self, axis=None, skipna=True, out=None):
...
         if not (
-            is_dask_collection(cond)
-            or is_dataframe_like(cond)
-            or is_series_like(cond)
-            or is_index_like(cond)
+            is_dask_collection(cond_res)
+            or is_dataframe_like(cond_res)
+            or is_series_like(cond_res)
+            or is_index_like(cond_res)
+            or callable(cond_res)
```
Comment (Oreilles) [inline dask/dataframe/core.py:3325]: Thanks for taking my review into account! I think this line should probably be removed then ? (the callable result shouldn't be a callable itself)

Comment (j-bennet) [reply dask/dataframe/core.py]: 👍

**Review by phofl — APPROVED** [review]

**Pull request merged**